### ISSUE TITLE
Discrete diffusion in the code

### ISSUE THREAD

**Issue by vadimkantorov** [OpenGVLab/DDPS#3]
Hi! Thanks a lot for releasing the codes :)

Do I understand correctly that the main model is in https://github.com/OpenGVLab/DDPS/blob/main/mmseg_custom/models/decode_heads/segformer_head_unet_fc_head_single_step.py#L128 ?

Am I correct that first the discrete class labels are embedded into real-valued vectors, put into the unet which outputs real-valued logits? (and that corruption is done in the discrete label space)

**Comment by Siddharth-Shrivastava7** [comment]
Hi @vadimkantorov , 

As far as I know, this codebase is similar to implementation of "Argmax Flows and Multinomial Diffusion:
Learning Categorical Distributions",  Reference: https://github.com/ehoogeboom/multinomial_diffusion/blob/9d907a60536ad793efd6d2a6067b3c3d6ba9fce7/segmentation_diffusion/layers/layers.py#L190

Hope this helps!
